### ISSUE TITLE
oneOf for properties

### ISSUE THREAD

**Issue by kondaurovDev** [eclipsesource/play-json-schema-validator#54]
Hi!
I figured out that you can use "oneOf", "anyOf" in that way:

```
{
    "type" : "object",
    "properties" : {
        "name" : { "type": "string"},
        "name2": { "type": "string"}
    },
    "oneOf" : [ 
        {
            "required": ["name"]
        },
        {
            "required": ["name2"]
        }
    ]
}
```

So this object fails:
```
{
"name": "bla",
"name": "blabla"
}
```

But this one succeed:
```
{
"name": "bla"
} 
```
or 
```
{
"name2": "blabla"
}
```

I've come to this from that post:
http://stackoverflow.com/questions/24023536/json-schema-how-do-i-require-one-field-or-another-or-one-of-two-others-but-no

I get error when i validate an empty object ({}) in yours demo web app:

```
[
    {
        "schemaPath": "#",
        "errors": {
            "matched": [
                "/oneOf/0",
                "/oneOf/1"
            ]
        },
        "msgs": [
            "Instance matches more than one schema"
        ],
        "value": {},
        "instancePath": ""
    }
]
```

Does your library support this feature? 
Thanks!

**Comment by edgarmueller** [comment]
This looks like a bug, I'll take care of this for the next release.

**Comment by edgarmueller** [comment]
@kondaurovDev Just to be on the same page: Your example about getting an error when validating an empty object is correct in that sense, that it is indeed invalid, but the error type is wrong. What the validator should actually return here, is a message that tells that none of the schemas has been matched. Also, I think the first example you gave
```
{
"name": "bla",
"name": "blabla"
}
```
will actually validate successfully.

**Comment by kondaurovDev** [comment]
@edgarmueller I'v attached wrong example, sorry, i meant this one:
```
{
"name": "bla",
"name2": "blabla"
}
```

But anyway, none of examples works. 
When i try: 
```
{
"name": "bla",
"name2": "blabla"
}

```
there is validation error:
```
[
    {
        "schemaPath": "#",
        "errors": {
            "matched": [
                "/oneOf/0",
                "/oneOf/1"
            ]
        },
        "msgs": [
            "Instance matches more than one schema"
        ],
        "value": {
            "name": "bla",
            "name2": "blabla"
        },
        "instancePath": ""
    }
]
```

It's good! But it fails when i try 
```
{
"name": "bla"
}
```

i get:
```
[
    {
        "schemaPath": "#",
        "errors": {
            "matched": [
                "/oneOf/0",
                "/oneOf/1"
            ]
        },
        "msgs": [
            "Instance matches more than one schema"
        ],
        "value": {
            "name": "bla"
        },
        "instancePath": ""
    }
]
```

Why does validator tell that there is more instance match more than one schema? 
It matches only one schema:
{
    "type" : "object",
    "properties" : {
        "name" : { "type": "string"},
        "name2": { "type": "string"}
    },
    "required": ["name"]
}

**Comment by edgarmueller** [comment]
Ok, I see, thanks for the clarification! The problem is that sub-schemas aren't merged correctly with the parent schema during validation involving oneOf/anyOf/allOf, which leads to these confusing follow-up errors. I'll add your examples as test cases and provide a fix for the next version 0.8.0. Thanks!

**Comment by edgarmueller** [comment]
Fixed in 0.8.0.
